src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -103,7 +118,7 @@ public bool Contains(T item)
 
         public OneOrMany<T> RemoveAll(T item)
         {
-            if (HasOne)
+            if (HasOneItem)
             {
                 return EqualityComparer<T>.Default.Equals(item, _one) ? default : this;
```

Comment:
this looks broken. the 'default' OneOrMany is not a valid state for the type (which may be something we want to fix indepenently. It will be a 'default' array for _many. Which will then return 'true' for 'HasOneItem'. We should be able to show a test where this breaks.